pandas/core/groupby.py

```diff
@@ -1365,6 +1365,62 @@ def nth(self, n, dropna=None):
 
     @Substitution(name='groupby')
     @Appender(_doc_template)
+    def ngroup(self, ascending=True):
+        """
+        Number each group from 0 to the number of groups - 1.
+
+        This is the enumerative complement of cumcount.  Note that the
+        numbers given to the groups match the order in which the groups
+        would be seen when iterating over the groupby object, not the
+        order they are first observed.
```

Comment:
this comment is good, add something like this to the docs (in groupy.rst) where you show an example